sycl/test-e2e/ESIMD/api/functional/ctors/ctor_fill_core.cpp

```diff
@@ -5,7 +5,8 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
-// RUN: %{build} -fsycl-device-code-split=per_kernel -o %t.out
+// TODO: remove fno-fast-math option once a compiler issue is resolved
```

Comment:
-fno-fast-math will not work with clang-cl based drivers.
Please see the solution for some similar issue here: https://github.com/intel/llvm/pull/9661

Also, please add more details to this TODO message and to the description of this PR, what kind of compiler issue exists now and this PR tried to solve or work-around.

Reply:
Thank you for adding the comment. It though still doesn't tell why results are incorrect with ffast-math.
Is it just slightly different results, or completely broken and irrelevant results?

Reply:
Usually such huge difference is the result of some long iterative error-accumulation process.
If I read the test correctly, it only runs ctor test for simd<T,1> and simd<T,8>. 
8 - is not too big number to accumulate an error, right?

Ok, maybe std::isnan() used in that test causes the problem with -ffast-math, where it can give false true or false. If so, then that part of the test needs more elaborate changes to adopt -ffast-math.

If you see what kind of problem each of tests has, then that is good to have that info in TODO comment to simplify future analysis. Right now from TODO it is unclear what is wrong with compiler. It rather looks like the test is designed such a way that doesn't work with -ffast-math because of potential vectorization or using std::isnan().